File: src/utils/nhs.js
import dotenv from 'dotenv';
import fetch from 'node-fetch';
import { scrapeArticleContent } from './scraper.js';
dotenv.config();

export const fetchHealthAZData = async (condition) => {
  const url = `https://api.nhs.uk/live-well/${encodeURIComponent(condition)}`;
  const options = {
    method: 'GET',
    headers: {
      'subscription-key': process.env.NHS_API_KEY, // Ensure this key is securely managed
      'Content-Type': 'application/json',
      'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3',
    },
  };

  try {
    const response = await fetch(url, options);
    if (!response.ok) {
      throw new Error(`Error: ${response.status}`);
    }
    const data = await response.json();
    const extractedData = extractUrlsAndHeadlines(data);
    // console.log(extractedData);
    // Randomly select an article to scrape
    const randomArticle = getRandomArticle(extractedData);
    const articleContent = await scrapeArticleContent(randomArticle.url);
    console.log(articleContent);
    // Optionally, write the extracted URLs and headlines to a JSON file
    // await fs.writeFile('healthData.json', JSON.stringify(extractedData, null, 2), 'utf-8');
    // console.log('Data saved to healthData.json');

    return articleContent;
  } catch (error) {
    console.error('Failed to fetch Health A-Z data:', error);
  }
};

// Function to extract URLs and headlines from the API response and store them in an array
function extractUrlsAndHeadlines(apiResponse) {
  const result = [];

  if (apiResponse.mainEntityOfPage && Array.isArray(apiResponse.mainEntityOfPage)) {
    apiResponse.mainEntityOfPage.forEach(mainEntity => {
      if (mainEntity.mainEntityOfPage && Array.isArray(mainEntity.mainEntityOfPage)) {
        mainEntity.mainEntityOfPage.forEach(webPageElement => {
          result.push({
            url: webPageElement.url,
            headline: webPageElement.headline
          });
        });
      }
    });
  }

  return result;
}

// Function to randomly select an article from the extracted data
function getRandomArticle(extractedData) {
  const randomIndex = Math.floor(Math.random() * extractedData.length);
  return extractedData[randomIndex];
}

// Example usage
// (async () => {
//   try {
//     await fetchHealthAZData('exercise');
//   } catch (error) {
//     console.error('Error running fetchHealthAZData:', error);
//   }
// })();

File: nhs.py
# @app.get("/get_condition")
# async def get_condition(condition: Optional[str] = None):
#     nhs_api_key = os.getenv("NHS_API_KEY")

#     if not nhs_api_key:
#         raise HTTPException(status_code=500, detail="NHS_API_KEY is not set")

#     if not condition or condition not in app.config.CONDITIONS:
#         condition = random.choice(app.config.CONDITIONS)

#     url = f"{app.config.NHS_API_BASE_URL}{requests.utils.quote(condition)}"
#     headers = {
#         "subscription-key": nhs_api_key,
#         "Content-Type": app.config.CONTENT_TYPE,
#         "User-Agent": app.config.USER_AGENT,
#     }

#     response = requests.get(url, headers=headers)

#     if response.status_code == 200:
#         extracted_data = extract_urls_and_headlines(response.json())
#         random_data = get_random_article(extracted_data)
#         return random_data
#     else:
#         raise HTTPException(status_code=response.status_code, detail="Failed to fetch data")

# def extract_urls_and_headlines(api_response):
#     result = []

#     if "mainEntityOfPage" in api_response and isinstance(api_response["mainEntityOfPage"], list):
#         for main_entity in api_response["mainEntityOfPage"]:
#             if "mainEntityOfPage" in main_entity and isinstance(main_entity["mainEntityOfPage"], list):
#                 for web_page_element in main_entity["mainEntityOfPage"]:
#                     result.append({
#                         "headline": web_page_element.get("headline", ""),
#                         "url": web_page_element["url"],
#                     })
#     return result

# def get_random_article(extracted_data):